docs/docs/icicle/primitives/merkle.md

```diff
@@ -1,3 +1,270 @@
-# Merkle tree
 
-TODO update for V3
\ No newline at end of file
+# Merkle Tree API Documentation
+
+## What is a Merkle Tree?
+
+A **Merkle tree** is a cryptographic data structure that allows for **efficient verification of data integrity**. It consists of:
+- **Leaf nodes**, each containing a piece of data.
+- **Internal nodes**, which store the **hashes of their child nodes**, leading up to the **root node** (the cryptographic commitment).
```

Comment:
stores

Reply:
"store" is correct here and shouldn't be plural